content/cloud/variables.md

```diff
@@ -0,0 +1,208 @@
+
+Let's start by [creating a new Spin application](/cloud/cli-reference#new) from a the Python template. Using the docs as a reference, we can perform the following:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin new http-py pw_checker
+Project description: A Spin app that validates a password
+HTTP base: /
+HTTP path: /...
+```
+
+## Adding Variables to an Application Component
+
+Next, let's add two variables to our application. Add a top level `[variables]` section to the application manifest (`spin.toml`) and define a `password` and `expected_password` variable, marking them as required. Now, add these variables to a component by referencing them in the `[component.config]` section in the `pw-checker` component. Instead of statically assigning the value of the config variables, we are referencing the dynamic variables with [mustache](https://mustache.github.io/)-inspired string templates. The resultant application manifest should look similar to the following:
+
+> Note: only components that explicitly use the variables in their configuration section will get access to them.
+
+<!-- @selectiveCpy -->
+```toml
+spin_manifest_version = "1"
+description = "A Spin app with a dynamically updatable "
+name = "pw_checker"
+trigger = { type = "http", base = "/" }
+version = "0.1.0"
+
+[variables]
+expected_password = { required = true }
+password = { required = true }
+
+[[component]]
+id = "pw-checker"
+source = "app.wasm"
+[component.trigger]
+route = "/..."
+[component.build]
+command = "spin py2wasm app -o app.wasm"
+[component.config]
+password = "{{ password }}"
+expected_password = "{{ expected_password }}"
+```
+
+## Using Variables in a Spin Application
+
+Now that we have defined our variables and exposed them to our component, lets implement our Spin application. The application should retrieve each variable and validate that the expected password matched the actual password. We will use the Spin `config_config` SDK to get the values. 
+
+<!-- @selectiveCpy -->
+```py
+from spin_http import Response
+from spin_config import config_get
+
+
+def handle_request(request):
+    expected = config_get("expected_password")
+    actual = config_get("password")
+    response = "Access denied"
+    if expected == actual:
+        response = "Access permitted"
+
+    return Response(200,
+                    {"content-type": "text/plain"},
+                    bytes(response, "utf-8"))
+
+```
+
+Lets build and run our application locally to test it out. We will use the [environment variable provider](../spin/dynamic-configuration.md#environment-variable-provider) to set the variable values locally.
+
+<!-- @selectiveCpy -->
+```console
+$ spin build
+$ SPIN_CONFIG_EXPECTED_PASSWORD="123" SPIN_CONFIG_
+PASSWORD="123" spin up
+Logging component stdio to ".spin/logs/"
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  pw-checker: http://127.0.0.1:3000 (wildcard)
+```
+
+Let's query our application to see if we are granted access.
```

Comment:
This is super subjective I know, but I'm starting to find the repetition of "let's" a bit grating.  The "doing this together" style is fine (and probably friendlier than my usual imperative style), but let's mix up the phrasi-- KATE YOU'VE GOT ME DOING IT NOW

Reply:
Can do. I'll make it less repetitive